hawkbit-repository/src/main/java/org/eclipse/hawkbit/repository/DeploymentManagement.java

```diff
@@ -349,27 +345,15 @@ private DistributionSetAssignmentResult assignDistributionSetToTargets(@NotNull
         targetIds.forEach(tIds -> targetRepository.setAssignedDistributionSet(set, System.currentTimeMillis(),
                 currentUser, tIds));
         targetIds.forEach(tIds -> targetInfoRepository.setTargetUpdateStatus(TargetUpdateStatus.PENDING, tIds));
-
-        final Map<String, Action> targetIdsToActions = actionRepository.save(targets.stream().map(t -> {
-            final Action tAction = new Action();
-            final TargetWithActionType targetWithActionType = targetsWithActionMap.get(t.getControllerId());
-            tAction.setActionType(targetWithActionType.getActionType());
-            tAction.setForcedTime(targetWithActionType.getForceTime());
-            tAction.setActive(true);
-            tAction.setStatus(Status.RUNNING);
-            tAction.setTarget(t);
-            tAction.setDistributionSet(set);
-            tAction.setRollout(rollout);
-            tAction.setRolloutGroup(rolloutGroup);
-            return tAction;
-        }).collect(Collectors.toList())).stream()
-                .collect(Collectors.toMap(a -> a.getTarget().getControllerId(), Function.identity()));
-
-        // create initial action status when action is created so we
-        // remember the initial
-        // running status because we will change the status of the action itself
-        // and with this action
-        // status we have a nicer action history.
+        final Map<String, Action> targetIdsToActions = actionRepository
+                .save(targets.stream().map(t -> createTargetAction(targetsWithActionMap, t, set, rollout, rolloutGroup))
+                        .collect(Collectors.toList()))
```

Comment:
Sorry Micha but I can't follow you. If I remove the colllect() in line 350 I get a "Type mismatch: cannot convert from Stream<Object> to Iterable<S>" because the save method is expecting an Iterable (e.g. list). The first collect is not for the result of the save method it is for the input after streamed. 

List<Action> actions =  targets.stream().map(t -> createTargetAction(targetsWithActionMap, t, set, rollout, rolloutGroup)).collect(Collectors.toList());
actionRepository.save(actions);

After saving we get a list back that then gets streamed again to add the controllerId as key for the map.